File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int i, j, k;
        int index = 0;
        
        for(int[] command : commands) {
            i = command[0];
            j = command[1];
            k = command[2];
            int[] result = Arrays.copyOfRange(array, i - 1, j);
            Arrays.sort(result);
            answer[index] = result[k-1];
            index++;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    var answer = [];

    for (let command of commands) {
        let i = command[0];
        let j = command[1];
        let k = command[2];
        
        let result = array.slice(i - 1, j);
        result.sort((a, b) => a - b);
        answer.push(result[k - 1]);
    }
    
    // 다른 풀이 1
    // for(let i = 0; i < commands.length; i++){
    //     let eachCommand = commands[i]
    //     let slice = array.slice(eachCommand[0] - 1, eachCommand[1])
    //           answer.push(slice.sort((a, b) => a - b)[eachCommand[2] - 1])
    // }

    // 다른 풀이2
    // for(let value of commands){
    //     answer.push(array.slice(value[0]-1, value[1]).sort((a,b)=>a-b)[value[2]-1]);
    // }
    
    // 다른 풀이3
    // answer = commands.map(a=>{
    //     return array.slice(a[0]-1,a[1]).sort((b,c)=>b-c)[a[2]-1];
    // })
    
    return answer;
}